modules/drivers/mongo/src/metabase/driver/mongo/util.clj

```diff
@@ -104,19 +133,16 @@
             authdb           (if (seq authdb)
                                authdb
                                dbname)
-            server-address   (mg/server-address host port)
-            credentials      (when user
-                               (mcred/create user authdb pass))
-            connect          (partial mg/connect server-address (build-connection-options :ssl? ssl, :additional-options additional-options))
-            conn             (if credentials
-                               (connect credentials)
-                               (connect))
-            mongo-connection (mg/get-db conn dbname)]
+            [mongo-client db] (->
+                                (if (= 2 (-> host frequencies (get \.))) ; if fqdn, use DNS SRV
```

Comment:
Could you please write down in comments or docstrings all the reasons we are doing thing the way we're doing them. (Especially why we use `connect` vs. `connect-srv`, and what those mean). Basically the stuff you explained to me as we were talking about the problem on Slack. That way in 2 years from now someone doesn't come in and have no clue why things are like this, assume it's duplicated code, and tear it out, or one of the community members doesn't decide to try Clojure for the first time and end up pulling their hair out trying to debug the wrong stuff for their connection.

Reply:
Docstrings in